test/keystore/persistence/EncryptedPersistence.test.ts

```diff
@@ -100,34 +98,43 @@ describe('EncryptedPersistence', () => {
 
     // Read the raw result, change one byte, write it back
     const rawResult = await persistence.getItem(TEST_KEY)!
-    rawResult![7] += 1
-    await persistence.setItem(TEST_KEY, rawResult!)
+    const parsedRawResult = SignedEciesCiphertext.fromBytes(rawResult!)
+    const newCiphertext = {
+      ...parsedRawResult.ciphertext,
+      mac: getRandomValues(new Uint8Array(32)),
+    }
+    const newData = await SignedEciesCiphertext.create(
+      newCiphertext,
+      privateKey
+    )
+    await persistence.setItem(TEST_KEY, newData.toBytes())
 
     // Expect an error if the ciphertext is tampered with
     await expect(encryptedPersistence.getItem(TEST_KEY)).rejects.toThrow(
-      'Bad MAC'
+      'Bad mac'
     )
   })
 
-  it('detects length modified ciphertext', async () => {
-    const data = getRandomValues(new Uint8Array(128))
+  it('detects bad signature', async () => {
     const persistence = new LocalStoragePersistence()
     const encryptedPersistence = new EncryptedPersistence(
       persistence,
       privateKey
     )
-
+    const data = getRandomValues(new Uint8Array(64))
     await encryptedPersistence.setItem(TEST_KEY, data)
-    // Read the raw result, change one byte, write it back
-    const rawResult = await persistence.getItem(TEST_KEY)!
-    // Add a byte to the rawResult
-    const newRawResult = new Uint8Array(rawResult!.length + 1)
-    newRawResult.set(rawResult!)
-    await persistence.setItem(TEST_KEY, newRawResult)
+    const encryptedBytes = await persistence.getItem(TEST_KEY)
+    const goodData = SignedEciesCiphertext.fromBytes(encryptedBytes!)
+    const signedBySomeoneElse = await SignedEciesCiphertext.create(
+      goodData.ciphertext,
+      (
+        await PrivateKeyBundleV1.generate()
+      ).identityKey
+    )
+    await persistence.setItem(TEST_KEY, signedBySomeoneElse.toBytes())
 
-    // Expect an error if the ciphertext is tampered with
-    await expect(encryptedPersistence.getItem(TEST_KEY)).rejects.toThrow(
-      'Invalid data length'
+    expect(encryptedPersistence.getItem(TEST_KEY)).rejects.toThrow(
+      'signature validation failed'
```

Comment:
nice conversion! Is it still worth having a "signed correctly, but encrypted incorrectly" test case?

Reply:
Good call. Will add.

Reply:
Added